nautobot/core/middleware.py

```diff
@@ -24,12 +32,37 @@ def header(self):
     def process_request(self, request):
 
         # Bypass middleware if remote authentication is not enabled
-        if not settings.REMOTE_AUTH_ENABLED:
+        if "nautobot.core.authentication.RemoteUserBackend" not in settings.EXTERNAL_AUTH_BACKENDS:
             return
 
         return super().process_request(request)
 
 
+class ExternalAuthMiddleware(AuthenticationMiddleware):
+    """
+    Custom implementation of Django's AuthenticationMiddleware used to set permissions for ephemeral users
+    """
+
+    def process_request(self, request):
+
+        # Bypass middleware if remote authentication is not enabled
+        if not settings.EXTERNAL_AUTH_BACKENDS:
```

Comment:
Why implement our own list of auth backends vs. inserting the desired backend to `AUTHENTICATION_BACKENDS`? Is it an attempt to eliminate the need for users to ever modify the core settings?

What appears to be missing is the authentication hooks for the external backends, or are you still working on that?

Reply:
Good catch, this needs to reflect my later decision to ditch overwriting existing vars.